File: app/src/main/java/com/example/engineeraitest/presentation/screen/main/adapter/ItemAdapter.java
package com.example.engineeraitest.presentation.screen.main.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.engineeraitest.R;
import com.example.engineeraitest.app.di.GlideApp;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ItemAdapter extends RecyclerView.Adapter<ItemAdapter.ItemViewHolder> {

    private List<String> mItemList = new ArrayList<>();

    @NotNull
    @Override
    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_user_item, parent, false);
        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ItemViewHolder holder, int position) {
        holder.bind(mItemList.get(position));
    }

    @Override
    public int getItemCount() {
        return mItemList.size();
    }

    public void setItemList(List<String> itemList) {
        mItemList.clear();
        mItemList.addAll(itemList);
        notifyDataSetChanged();
    }

    class ItemViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.iv_item)
        ImageView mItemImageView;

        ItemViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        void bind(String item) {
            GlideApp.with(itemView.getContext())
                    .load(item)
                    .placeholder(R.color.color_placeholder)
                    .error(R.color.color_placeholder)
                    .into(mItemImageView);
        }
    }
}

File: app/src/main/java/com/example/engineeraitest/presentation/screen/main/adapter/ItemAdapter.kt
package com.example.engineeraitest.presentation.screen.main.adapter

import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.engineeraitest.R
import com.example.engineeraitest.app.di.GlideApp
import com.example.engineeraitest.presentation.tools.layoutInflater
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_user_item.*

class ItemAdapter : RecyclerView.Adapter<ItemAdapter.ItemViewHolder>() {

    private val mItemList = ArrayList<String>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        return ItemViewHolder(parent.context.layoutInflater.inflate(R.layout.item_user_item, parent, false))
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        holder.bind(mItemList[position])
    }

    override fun getItemCount(): Int {
        return mItemList.size
    }

    fun setItemList(itemList: List<String>) {
        mItemList.addAll(itemList)
        notifyDataSetChanged()
    }

    inner class ItemViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView),
        LayoutContainer {

        fun bind(item: String) {

            GlideApp.with(containerView.context)
                .load(item)
                .placeholder(R.color.color_placeholder)
                .error(R.color.color_placeholder)
                .into(iv_item)
        }
    }
}